manta-crypto/src/arkworks/ff.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Arkworks Finite Field Backend
+
+use manta_util::{byte_count, into_array_unchecked};
+
+#[doc(inline)]
+pub use ark_ff::*;
+
+/// Implements a fallible conversion from `F` to `$type`.
+macro_rules! field_try_into {
+    ($($name:ident => $type:tt),* $(,)?) => {
+        $(
+            #[doc = "Tries to convert a field element `x` into an integer of type `"]
+            #[doc = stringify!($type)]
+            #[doc = "`."]
+            #[inline]
+            pub fn $name<F>(x: F) -> Option<$type>
+            where
+                F: PrimeField,
+            {
+                if x < F::from(2u8).pow([$type::BITS as u64]) {
```

Comment:
What would happen if `$type::BITS as u64` is quite large such that `F::from(2u8).pow([$type::BITS as u64])` becomes a small integer after module?

We should have a comments to say `$type::BITS as u64` should be less than bits of `F`.

Reply:
Yeah, I can add a comment on the macro. It doesn't make sense to add it to the function.